src/openrct2/Date.cpp

```diff
@@ -0,0 +1,129 @@
+#pragma region Copyright (c) 2014-2017 OpenRCT2 Developers
+/*****************************************************************************
+* OpenRCT2, an open source clone of Roller Coaster Tycoon 2.
+*
+* OpenRCT2 is the work of many authors, a full list can be found in contributors.md
+* For more information, visit https://github.com/OpenRCT2/OpenRCT2
+*
+* OpenRCT2 is free software: you can redistribute it and/or modify
+* it under the terms of the GNU General Public License as published by
+* the Free Software Foundation, either version 3 of the License, or
+* (at your option) any later version.
+*
+* A full copy of the GNU General Public License can be found in licence.txt
+*****************************************************************************/
+#pragma endregion
+
+#include "core/Guard.hpp"
+#include "core/Math.hpp"
+#include "Date.h"
+
+using namespace OpenRCT2;
+
+constexpr sint32 MONTH_TICKS_INCREMENT = 4;
+constexpr sint32 MASK_WEEK_TICKS = 0x3FFF;
+constexpr sint32 MASK_FORTNIGHT_TICKS = 0x7FFF;
+constexpr sint32 MASK_MONTH_TICKS = 0xFFFF;
+
+Date::Date()
+{
+}
+
+Date::Date(uint32 monthsElapsed, uint16 monthTicks)
+    : _monthsElapsed(monthsElapsed),
```

Comment:
> Is this not describing behaviour rather than syntax?

It is.

> If so, why does gcc allow it?

GCC will reorder the initialisation automatically for you to match standard, but it will generate a warning when it does so. With `-Werror` it results in failed CI run. From earlier job with defective code:

```
In file included from /home/travis/build/OpenRCT2/OpenRCT2/src/openrct2/Date.cpp:19:0:
/home/travis/build/OpenRCT2/OpenRCT2/src/openrct2/Date.h: In constructor 'OpenRCT2::Date::Date(uint32, uint16)':
/home/travis/build/OpenRCT2/OpenRCT2/src/openrct2/Date.h:35:33: error: 'OpenRCT2::Date::_monthsElapsed' will be initialized after [-Werror=reorder]
         uint32 _monthsElapsed = 0;
                                 ^
/home/travis/build/OpenRCT2/OpenRCT2/src/openrct2/Date.h:34:30: error:   'uint16 OpenRCT2::Date::_monthTicks' [-Werror=reorder]
         uint16 _monthTicks = 0;
                              ^
/home/travis/build/OpenRCT2/OpenRCT2/src/openrct2/Date.cpp:32:1: error:   when initialized here [-Werror=reorder]
 Date::Date(uint32 monthsElapsed, uint16 monthTicks)
 ^
cc1plus: all warnings being treated as errors
```